### ISSUE TITLE
Why is the training accuracy not increasing while implementing MemN2N ?

### ISSUE THREAD

**Issue by dasgauranga2** [keras-team/keras#12667]
I am trying to implement the MemN2N from here. I tried implementing the code using the Functional API of Keras.I tried changing the optimizer but, I got the same results.

Below is my implementation:

```
input_sequence = Input((story_maxlen,))
question = Input((query_maxlen,))

input_mem_encoding = Embedding(input_dim=vocab_size, output_dim=64, input_length=story_maxlen)(input_sequence)

query_encoding = Embedding(input_dim=vocab_size, output_dim=64, input_length=query_maxlen)(question)

softmax_input_query = dot([input_mem_encoding,query_encoding],axes=(2,2))
softmax_input_query = Activation('softmax')(softmax_input_query)

input_sm_encoding = Embedding(input_dim=vocab_size, output_dim=64, input_length=story_maxlen)(input_sequence)
input_sm_encoding = Permute((2, 1))(input_sm_encoding)

output = dot([softmax_input_query,input_sm_encoding],axes=(1,2))
output = Dropout(0.3)(output)

answer = add([output, query_encoding])
answer = Dropout(0.3)(answer)
answer = LSTM(32)(answer)
answer = Dropout(0.3)(answer)
answer = Dense(vocab_size, activation='softmax')(answer)

model = Model([input_sequence,question],answer)

model.compile(optimizer='adam', loss='sparse_categorical_crossentropy', metrics=['accuracy'])

model.fit([inputs_train, queries_train], answers_train,
           batch_size=32,
           nb_epoch=120,
           validation_data=([inputs_test, queries_test], answers_test))
```
When I fit the model the training accuracy increases upto 40% and then doesn't increases further.
